### DISCUSSION TITLE
Nextflow Run from Git: Get the Tag into Script/Pipeline

### DISCUSSION THREAD

**Discussion by tjdurant** [nextflow-io/nextflow#2995] [Q&A] [answered]
Hi All,

When running pipelines, our group runs them from a gitlab repo using a tag:
```
nextflow run <private repo> -profile docker --batch_id <batch_id> -r <tag_version>
```

We would like to pull the `<tag_version>` into the pipeline and output it in a file. 

Does anyone know if that is possible?

Or, what do other people do? I know we can update the config manifest to include the version, but we've had instances were developers have forgotten to update the config file. 

Any thoughts/comments/suggestions are welcome.

Thanks!
Tom

**Comment by pditommaso** [comment, marked as answer]
You can use `-with-report` option that track that info. 

<img width="1423" alt="Screenshot 2022-06-27 at 16 54 05" src="https://user-images.githubusercontent.com/816968/175970543-634cd418-cc7a-4eec-9b8b-555516fb4b7e.png">


Even better you can use the `-with-tower` option that will upload those info to https://tower.nf/

**Reply by tjdurant** [reply]
thank you @pditommaso!

One follow-up question, do you (or other community members), have recommendations on how to pass this report to the pipeline output directory if the output paths are defined in the `nextflow.config` file?

**Reply by pditommaso** [reply]
make a wrapper script to handle it!

**Reply by tjdurant** [reply]
thanks, any good examples out there that you know of?

**Reply by pditommaso** [reply]
```
#!/bin/bash
nextflow run <YOUR PIPELINE> -with-report 
cp report.html /some/where
```

**Comment by tjdurant** [comment]
We ended up defining the path to the report in the report [scope](https://www.nextflow.io/docs/latest/config.html#:~:text=Scope%20report,%EF%83%81)

```
report {
    file = "${params.project_root_dir}/COVID_SEQ_${params.pipeline_version}/BATCH_${params.batch_id}_${params.bed_version}/pipeline_output_summary/nextflow_pipeline_report.html"
}
```

**Comment by stevekm** [comment]
This seems to work when the pipeline is started from a remote repository (`nextflow run somerepo/somepipeline`) but it does not seem to work if Nextflow is run directly from within the repo (`nextflow run main.nf`)
would be helpful if Nextflow could report the output from something like `git describe` (e.g. `1.16-6-g934775a`) and use that within the pipeline for reporting
